cmd/pj-rehearse/main_test.go

```diff
@@ -152,15 +156,31 @@ func getRehersalsHelper(logger *logrus.Entry, prNumber int) ([]*prowconfig.Presu
 		return nil, fmt.Errorf("failed to load ci-operator config: %v", err)
 	}
 
+	masterTemplates, err := config.GetTemplates(masterTemplatesPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load master templates: %v", err)
+	}
+	candidateTemplates, err := config.GetTemplates(candidateTemplatesPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load candidate templates: %v", err)
+	}
+
 	changedPresubmits := diffs.GetChangedPresubmits(prowConfig, prowPRConfig, logger)
 	if len(changedPresubmits) == 0 {
 		return nil, fmt.Errorf("Empty changedPresubmits was not expected")
 	}
 
+	loggers := rehearse.Loggers{Job: logger, Debug: logger}
+
 	changedCiopConfigs, affectedJobs := diffs.GetChangedCiopConfigs(ciopMasterConfig, ciopPrConfig, logger)
 	changedPresubmits.AddAll(diffs.GetPresubmitsForCiopConfigs(prowPRConfig, changedCiopConfigs, logger, affectedJobs))
 
-	rehearsals := rehearse.ConfigureRehearsalJobs(changedPresubmits, ciopPrConfig, prNumber, rehearse.Loggers{Job: logger, Debug: logger}, false, nil)
+	changedTemplates := diffs.GetChangedTemplates(masterTemplates, candidateTemplates, logger)
+	templatesWithClusterTypes := rehearse.GetTemplatesWithClusterTypes(changedTemplates)
+	presubmitsWithChangedTemplates := rehearse.AddRandomJobsForChangedTemplates(templatesWithClusterTypes, changedTemplates, prowPRConfig.JobConfig.Presubmits, loggers, prNumber)
+	changedPresubmits.AddAll(presubmitsWithChangedTemplates)
+
+	rehearsals := rehearse.ConfigureRehearsalJobs(changedPresubmits, ciopPrConfig, prNumber, loggers, true, changedTemplates)
```

Comment:
It has been mentioned before, but the fact that this is a copy of `main.go` is really bad for maintenance. Can we at least call that code from here?

Reply:
I really think that we can drop all of the `main_test.go` code. The integration tests that we have are sufficient for this purpose.
@petr-muller what do you think? Can we kill that code?

Reply:
> I really think that we can drop all of the main_test.go code. The integration tests that we have are sufficient for this purpose.

SGTM

Reply:
addressed [here](https://github.com/openshift/ci-operator-prowgen/pull/125)